### PR TITLE
Added additional parameters void pointer and sim pointer to particle struct

### PR DESCRIPTION
**Pull request by dtamayo**
I think this works, but wait for Travis

### CODE REVIEW

File: src/rebound.h [hunk 1]
```diff
@@ -107,6 +107,8 @@ struct reb_particle {
 	double lastcollision;		///< Last time the particle had a physical collision.
 	struct reb_treecell* c;		///< Pointer to the cell the particle is currently in.
 	int id;				///< Unique id to identify particle.
+	void* ap;			///< Functionality for externally adding additional properties to particles.
+	struct reb_simulation* sim;
```
Comment (hannorein) [inline src/rebound.h]: How about a documentation string here?

**Comment by hannorein** [comment]
Looks simple and clean!

**Comment by dtamayo** [comment]
Sorry, I had that in there before and forgot to put back in.  I also added something for _sim in Particle...I thought it was better to document that it shouldn't be used, rather than letting people wonder and try it?

**Comment by hannorein** [comment]
Agreed!

**Pull request merged**